src/k8s/K8sWorkerController.cpp

```diff
@@ -0,0 +1,142 @@
+        "'" + std::string(service->metadata->name) + "', " +
+        "'" + std::string(service->spec->cluster_ip) + "', " +
+        std::to_string(workerId) + ")";
+    int status = metadb.runInsert(insertQuery);
+    if (status == -1) {
+        controller_logger.error("Worker " + std::to_string(workerId) + " database insertion failed");
+    }
+}
+
+void K8sWorkerController::deleteWorker(int workerId) {
+    v1_status_t *status = this->interface->deleteJasmineGraphWorkerDeployment(workerId);
+    if (status != nullptr && status->code == 0) {
+        controller_logger.info("Worker " + std::to_string(workerId) + " deployment deleted successfully");
+    } else {
+        controller_logger.error("Worker " + std::to_string(workerId) + " deployment deletion failed");
+    }
+
+    v1_service_t *service = this->interface->deleteJasmineGraphWorkerService(workerId);
+    if (service != nullptr && service->metadata != nullptr && service->metadata->name != nullptr) {
+        controller_logger.info("Worker " + std::to_string(workerId) + " service deleted successfully");
+    } else {
+        controller_logger.error("Worker " + std::to_string(workerId) + " service deletion failed");
+    }
+
+    std::string deleteQuery = "DELETE FROM worker WHERE idworker = " + std::to_string(workerId);
+    metadb.runUpdate(deleteQuery);
+}
+
+int K8sWorkerController::attachExistingWorkers() {
+    v1_deployment_list_t *deployment_list =
+        this->interface->getDeploymentList(strdup("deployment=jasminegraph-worker"));
+
+    if (deployment_list && deployment_list->items->count > 0) {
+        listEntry_t *listEntry = NULL;
+        v1_deployment_t *deployment = NULL;
+        list_ForEach(listEntry, deployment_list->items) {
+            deployment = static_cast<v1_deployment_t *>(listEntry->data);
+            list_t *labels = deployment->metadata->labels;
+            listEntry_t *label = NULL;
+
+            list_ForEach(label, labels) {
+                auto *pair = static_cast<keyValuePair_t *>(label->data);
+                v1_service_t *service;
+                if (strcmp(pair->key, "workerId") == 0) {
+                    int workerId = std::stoi(static_cast<char *>(pair->value));
+                    v1_service_list_t *service_list = this->interface->getServiceList(
+                        strdup(("service=jasminegraph-worker,workerId="
+                            + std::to_string(workerId)).c_str()));
+
+                    if (!service_list || service_list->items->count == 0) {
+                        service = this->interface->createJasmineGraphWorkerService(workerId);
+                    } else {
+                        service = static_cast<v1_service_t *>(service_list->items->firstEntry->data);
+                    }
+
+                    std::string insertQuery =
+                        "INSERT INTO worker (host_idhost, server_port, server_data_port, name, ip, idworker) "
```

Comment:
There is a confusion between the server_port and server_data port here. Why do we use fronted port and backend ports here as the values.

Reply:
Yes. This creates a confusion. My intention is to keep the same ports exposed in all master and workers.
Anyway we can shift the ports to use `JASMINEGRAPH_INSTANCE_PORT = 7780` and `JASMINEGRAPH_INSTANCE_DATA_PORT = 7781` instead

Reply:
Updated the PR in this approach

Reply:
Ok